### PR TITLE
Improve the code quality ruff action

### PR DESCRIPTION
**Pull request by jiridanek**
* clone the bare minimum
* cache installed poetry
* cache poetry virtualenv
* run poetry check --lock

### CODE REVIEW

**Review by jstourac — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

File: .github/workflows/code_quality.yaml [hunk 1]
```diff
@@ -84,15 +82,33 @@ jobs:
...
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
+      - name: Cache poetry in ~/.local
+        uses: actions/cache@v3
+        id: cached-home-local
+        with:
+          path: ~/.local
+          key: ${{ runner.os }}-local-${{ env.poetry_version }}
```
Comment (bdattoma) [inline .github/workflows/code_quality.yaml:97]: what is the benefit of this step?

File: .github/workflows/code_quality.yaml [hunk 2]
```diff
@@ -84,15 +82,33 @@ jobs:
...
+        with:
+          cache: 'poetry'
+
+      - run: poetry check --lock
+
       - name: Install ruff
-        run: |
-          pip install poetry
-          poetry install --only dev
+        run: poetry install --only=dev --sync
```
Comment (bdattoma) [inline .github/workflows/code_quality.yaml:112]: why installing once again?

File: .github/workflows/code_quality.yaml (hunk 1 above)
Comment (jiridanek) [reply .github/workflows/code_quality.yaml:97]: It allows skipping pip install poetry (the next step) if restoring cache succeeds. Restoring poetry from cache this way takes a few seconds less than installing with pip every time.

So, it saves 3 to 5 seconds off of the job runtime.

File: .github/workflows/code_quality.yaml (hunk 2 above)
Comment (jiridanek) [reply .github/workflows/code_quality.yaml:112]: The previous install only got us the poetry executable itself. Now I use poetry to install the ods-ci dependencies

Comment (bdattoma) [reply .github/workflows/code_quality.yaml:112]: oh right, stupid me, sorry.

**Pull request merged**